src/interactivity/win32/windowio.cpp

```diff
@@ -167,6 +167,27 @@ void HandleKeyEvent(const HWND hWnd,
         // --- END LOAD BEARING CODE ---
     }
 
+    // - Certain applications like PowerToys and AutoHotKey, due to their keyboard
+    //   remapping feature, send us key events (WM_KEYUP, WM_KEYDOWN, etc.) using
+    //   SendInput(), whose keycode or scancode values could be outside of the
+    //   valid range. We need to filter those events out, as some clients will
+    //   incorrectly translate them to an ascii NULL character.
+    // - If scan code is 0, we can still infer it from the virtual key code as long
+    //   as the virtual key code is valid (0 < vKey < 0xFF).
+    if (VirtualScanCode == 0) [[unlikely]]
+    {
+        // try to infer scancode from virtual keycode
+        auto FullVirtualScanCode = gsl::narrow_cast<WORD>(OneCoreSafeMapVirtualKeyW(VirtualKeyCode, MAPVK_VK_TO_VSC_EX));
+        VirtualScanCode = LOBYTE(FullVirtualScanCode);
+        // if we still don't have a scancode, return.
+        if (VirtualScanCode == 0)
+        {
+            return;
+        }
+        // Otherwise, set 'enhanced' bit if necessary and continue.
+        ControlKeyState |= (HIBYTE(FullVirtualScanCode) == 0xE0) && ENHANCED_KEY;

```

Comment:
Is this supposed to say `&` instead of `&&`?

Reply:
It is suppose to bitwise-or `ControlKeyState` with `ENHANCED_KEY` if first condition is true, or else with 0.

Is it doing anything else? Sometimes I just couldn't read my own code. 😂🥲

Reply:
Changed the statement a bit so the intent is clear. 😄